public/components/panel/studioPanel/studioPanel.js

```diff
@@ -69,20 +72,40 @@ class StudioPanel extends React.Component {
    * @param {GridBeatObject} beatObject
    */
   handleSaveBeatToDatabase = beatObject => {
-    // Important!: Should show progress status in overlay, should cancel out overlay on error
```

Comment:
Your code is pretty barren of comments which might be a code smell considering how large its getting, why did we remove these?

Reply:
These were a pointer to remind me of what I intended to code here. The function itself is just a setState and a lot of the comment is actually done elsewhere. Like from reading this:
```
this.setState({
      currentSavePromptInfo: {
        title: 'Save Beat',
        shouldShowPrompt: true,
        onSaveComplete: ...,
      },
})
```
you can infer that I'm providing the details of how the pop-up should look like and what happens onSaveComplete.